src/lib/util/pair.c

```diff
@@ -2394,3 +2394,28 @@ bool pair_matches_da(void *item, void *uctx)
 	printf("%s %s\n", vp->da->name, da->name);
 	return da == vp->da;
 }
+
+/** Wrapper function to call fr_cursor_filter_next() filtering by da
+ *
+ * To be used where the da being filtered by is declared as a const
+ *
+ * @param[in] cursor pointing to the list being searched
+ * @param[in] da to match
+ *
+ * @return
+ *	- next item in the list matching the given da
+ *	- NULL if no match found
+ */
+void *fr_cursor_filter_next_by_da(fr_cursor_t *cursor, const fr_dict_attr_t *uctx)
```

Comment:
this could be `... fr_cursor_filter_next_by_da(fr_cursor_t *cursor, fr_dict_attr_t const *da)`

i.e. we put `const` after the type, and there's no need to use `uctx` here.  Just use `da`.

The reason to keep with `uctx` is if the function signature is required to take `void*uctx`  We then cast that to the correct type inside of the function.

Reply:
unconsting da is wrong here.  The fr_cursor_filter_next code should really take a const and do the unconsting.

Reply:
Reason for `const` after the type is so that it consistently applies to the thing to the left of it.